tests/tests_unit/test_api/test_datapoints.py

```diff
@@ -1260,6 +1260,32 @@ def test_insert_dataframe_with_nans(self):
         with pytest.raises(AssertionError, match="contains NaNs"):
             DPS_CLIENT.insert_dataframe(df)
 
+    def test_insert_dataframe_with_dropna(self, mock_post_datapoints):
+        import pandas as pd
+
+        timestamps = [1500000000000, 1510000000000, 1520000000000, 1530000000000]
+        df = pd.DataFrame(
+            {"123": [1, 2, None, 4], "456": [5.0, 6.0, 7.0, 8.0]},
+            index=[utils._time.ms_to_datetime(ms) for ms in timestamps],
+        )
+        res = DPS_CLIENT.insert_dataframe(df, dropna=True)
```

Comment:
test the exception as well on dropna=False?

Reply:
It defaults to this in `test_insert_dataframe_with_nans`, do you want to see another test for it explicitly?

Reply:
no that's fine